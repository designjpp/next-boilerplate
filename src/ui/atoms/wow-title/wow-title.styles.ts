import { css } from '@emotion/css';
import xw from 'xwind';

const wowTitleStyles = xw`inline-block text-center text-transparent opacity-50 prose-wow-xl bg-clip-text bg-gradient-to-r from-blue to-yellow`;

export const wowTitleClassName = css`
	${wowTitleStyles}
`;
