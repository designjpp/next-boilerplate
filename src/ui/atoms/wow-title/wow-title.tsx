import { wowTitleClassName } from './wow-title.styles';

import type { FC } from 'react';

interface WowTitleInterface {
	as?: string;
}

export const WowTitle: FC<WowTitleInterface> = ({
	as: asComponent,
	children,
}) => {
	const CustomElement = asComponent || 'span';

	return (
		// @ts-ignore
		<CustomElement className={wowTitleClassName}>{children}</CustomElement>
	);
};
