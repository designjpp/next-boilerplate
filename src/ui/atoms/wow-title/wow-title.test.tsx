import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { create } from 'react-test-renderer';

import { WowTitle } from './wow-title';

describe('<WowTitle>', () => {
	it('should match snapshot', async () => {
		const component = create(<WowTitle>Wow</WowTitle>).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should match snapshot given another element', async () => {
		const component = create(<WowTitle as='p'>Wow</WowTitle>).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<WowTitle>Wow</WowTitle>);
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});
});
