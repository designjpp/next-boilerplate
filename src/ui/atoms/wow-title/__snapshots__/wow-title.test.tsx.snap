// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WowTitle> should match snapshot 1`] = `
.emotion-0 {
  --tw-gradient-from: #3AB3DA;
  --tw-gradient-stops: var(--tw-gradient-from),var(--tw-gradient-to, rgba(58, 179, 218, 0));
  --tw-gradient-to: #FFC93D;
  display: inline-block;
  text-align: center;
  color: transparent;
  opacity: 0.5;
  font-family: sans-serif;
  font-family: Sen;
  font-size: 5rem;
  line-height: 6rem;
  font-weight: 700;
  -webkit-background-clip: text;
  background-clip: text;
  background-image: linear-gradient(to right, var(--tw-gradient-stops));
}

<span
  className="emotion-0"
>
  Wow
</span>
`;
